inductiva/tasks/task.py

```diff
@@ -153,14 +155,14 @@ def set_output_class(self, output_class):
         """Set the output class of the task."""
         self._output_class = output_class
 
-    def get_output(self, output_dir=None):
+    def get_output(self, output_dir: Optional[types.Path] = None):
         """Get the output of the task.
 
         Returns:
             The output of the task.
         """
         self.wait()
-        output_dir = self.download_outputs(output_dir)
+        output_dir = self.download_outputs(output_dir=output_dir)
 
         if self._output_class:
             return self._output_class(output_dir)
```

Comment:
Quick question: if you don't uncompress, what happens here?
Does the output_class know how do deal with compressed formats?
Do you need to pass the information about things having been decompressed or not to the _outout_class so that is know that it has to decompress on demand.
Do you see my point?

@IvanPombo @begaiym-k wdyt?

What happens if we don't decompress the files and pass the output dir to one of these output classes?

Reply:
It won't work, the output classes do not have a notion to decompress the information if it's zipped.

From a user perspective, it does not make much sense to change the argument to keep compressed if he wants to perform their after post-processing. But either way, this possibility needs to be accounted for on the output class. @cristianacarpinteiro can you own this?

In my view, it may be preferable to create a general output class with this feature and the general initialisation with `sim_output_dir`. Thereafter, all output classes inherit from it and just perform the specific renders with things already unpacked.

Reply:
In that case, should we have that flag there?
Do you see my point?
Why are we letting a broken feature go to the hands of the user.

Now: having compressed files could make sense, but it just makes us have a lot more work. But if we are not having that work now, then we can't let this go to the hands of the user.

I need everyone to detect this issues.

Reply:
Yes, we can still have this flag here. The get_output() makes sure that the files are uncompressed. This is the way the user will fetch the Output class from the scenario. So it is not broken and can stay within the code. However, I misunderstood and was thinking the task.get_output() had the uncompressed flag as well.


The download_outputs is for more general stuff, when ones to download all files or other ones not considered in the task.get_output().